Havtorn/Source/Engine/ECS/Systems/RenderSystem.cpp

```diff
@@ -238,19 +248,45 @@ namespace Havtorn
 
 		for (U64 i = 0; i < spriteComponents.size(); i++)
 		{
-			const STransform2DComponent& transform2DComp = transform2DComponents[i];
-			if (!transform2DComp.IsInUse)
-				continue;
-
 			const SSpriteComponent& spriteComp = spriteComponents[i];
 			if (!spriteComp.IsInUse)
 				continue;
 
-			SRenderCommand command;
-			command.Transform2DComponent = transform2DComp;
-			command.SpriteComponent = spriteComp;
-			command.Type = ERenderCommandType::ForwardTransparency;
-			RenderManager->PushRenderCommand(command);
+			RenderManager->AddSpriteToInstancedUVRectRenderList(spriteComp.TextureIndex, spriteComp.UVRect);
+			RenderManager->AddSpriteToInstancedColorRenderList(spriteComp.TextureIndex, spriteComp.Color.AsVector4());
+
+			const STransformComponent& transformComp = transformComponents[i];
+			const STransform2DComponent& transform2DComp = transform2DComponents[i];
+
+			if (transformComp.IsInUse)
+			{
+				if (!RenderManager->IsSpriteInInstancedWSTransformRenderList(spriteComp.TextureIndex)) 
+				{
+					// NR: Don't push a command every time
+					SRenderCommand command;
+					command.SpriteComponent = spriteComp;
+					command.Type = ERenderCommandType::GBufferSpriteInstanced;
+					RenderManager->PushRenderCommand(command);
+				}
+
+				RenderManager->AddSpriteToInstancedWSTransformRenderList(spriteComp.TextureIndex, transformComp.Transform.GetMatrix());
```

Comment:
In reference to my previous comment on line 36.
Hmmm Maybe it's a bit too much? <3

```suggestion
				RenderManager->AddSpriteToInstancedWorldSpaceTransformRenderList(spriteComp.TextureIndex, transformComp.Transform.GetMatrix());
```